modules/gateway-task/variables.tf

```diff
@@ -0,0 +1,240 @@
+  description = "consul-ecs Docker image."
+  type        = string
+  default     = "public.ecr.aws/hashicorp/consul-ecs:0.4.1-dev"
+}
+
+variable "envoy_image" {
+  description = "Envoy Docker image."
+  type        = string
+  default     = "envoyproxy/envoy-alpine:v1.20.2"
+}
+
+variable "log_configuration" {
+  description = "Task definition log configuration object (https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_LogConfiguration.html)."
+  type        = any
+  default     = null
+}
+
+variable "retry_join" {
+  description = "Arguments to pass to -retry-join (https://www.consul.io/docs/agent/options#_retry_join). This or `consul_server_service_name` must be set."
+  type        = list(string)
+}
+
+variable "consul_http_addr" {
+  description = "Consul HTTP Address. Required when using the IAM Auth Method to obtain ACL tokens."
+  type        = string
+  default     = ""
+}
+
+variable "consul_https_ca_cert_arn" {
+  description = "The ARN of the Secrets Manager secret containing the CA certificate for Consul's HTTPS interface."
+  type        = string
+  default     = ""
+}
+
+variable "client_token_auth_method_name" {
+  description = "The name of the Consul Auth Method to login to for client tokens."
+  type        = string
+  default     = "iam-ecs-client-token"
+}
+
+variable "service_token_auth_method_name" {
+  description = "The name of the Consul Auth Method to login to for service tokens."
+  type        = string
+  default     = "iam-ecs-service-token"
+}
+
+variable "tags" {
+  description = "A map of tags to add to all resources."
+  type        = map(string)
+  default     = {}
+}
+
+variable "tls" {
+  description = "Whether to enable TLS for the mesh-task for the control plane traffic."
+  type        = bool
+  default     = false
+}
+
+variable "consul_server_ca_cert_arn" {
+  description = "The ARN of the Secrets Manager secret containing the Consul server CA certificate for Consul's internal RPC."
+  type        = string
+  default     = ""
+}
+
+variable "gossip_key_secret_arn" {
+  description = "The ARN of the Secrets Manager secret containing the Consul gossip encryption key."
+  type        = string
+  default     = ""
+}
+
+variable "acls" {
+  description = "Whether to enable ACLs for the gateway task."
+  type        = bool
+  default     = false
+}
+
+variable "consul_datacenter" {
+  type        = string
+  description = "The name of the Consul datacenter the client belongs to."
+  default     = "dc1"
+}
+
+variable "consul_agent_configuration" {
+  type        = string
+  description = "The contents of a configuration file for the Consul Agent in HCL format."
+  default     = ""
+}
+
+variable "kind" {
+  description = "Gateway kind."
+  type        = string
+
+  validation {
+    error_message = "Gateway kind must be one of 'mesh-gateway', 'terminating-gateway', 'ingress-gateway'."
+    condition     = contains(["mesh-gateway", "terminating-gateway", "ingress-gateway"], var.kind)
```

Comment:
What do you think about only allowing `mesh-gateway` for now?

Reply:
Yes, that is consistent with the change that we made to the `consul-ecs` config schema. It only allows `mesh-gateway`.